intro and see also for export parts (#31599)
mdn/content: files/en-us/web/html/global_attributes/exportparts/index.md

The **`exportparts`** [global attribute](/en-US/docs/Web/HTML/Global_attributes) allows you to select and style elements existing in nested {{Glossary("shadow tree", "shadow trees")}}, by exporting their `part` names.

The shadow tree is an isolated structure where identifiers, classes, and styles cannot be reached by selectors or queries belonging to a regular DOM. There are two HTML attributes that can be applied to shadow tree elements that enable targeting CSS styles from outside to shadow tree: `part` and `exportparts`.

> The global [`part`](/en-US/docs/Web/HTML/Global_attributes#part) attribute makes a shadow tree element visible to its parent DOM. Using the `part` name as the parameter, the {{CSSxRef("::part", "::part()")}} pseudo-element can be used to apply CSS styles to elements in the shadow tree from outside of it. However, the `::part()` pseudo-element is only visible to the parent DOM. This means that when a shadow tree is nested, the parts are not visible to any ancestors other than the direct parent. The `exportparts` attribute solves this limitation.

Author: ```suggestion
The global [`part`](/en-US/docs/Web/HTML/Global_attributes#part) attribute makes a shadow tree element visible to its parent DOM. A `part` name is used as the parameter of the {{CSSxRef("::part", "::part()")}} pseudo-element. In this way, you can apply CSS styles to elements in the shadow tree from outside of it. However, the `::part()` pseudo-element is only visible to the parent DOM. This means that when a shadow tree is nested, the parts are not visible to any ancestors other than the direct parent. The `exportparts` attribute solves this limitation.
```

> The global [`part`](/en-US/docs/Web/HTML/Global_attributes#part) attribute makes a shadow tree element visible to its parent DOM. The {{CSSxRef("::part", "::part()")}} pseudo-element, with the part name as the parameter, can then be used to apply CSS styles to elements in the shadow tree from outside the shadow tree. As the `::part()` pseudo-element is only visible to the parent DOM, when a shadow tree is nested, the parts are not visible to any ancestors other than the direct parent. The `exportparts` attribute solves this limitation as it enables visibility into nested shadow trees, .

Reviewer: - The second sentence is introducing the pseudo-element abruptly. I recommend that we start the sentence with `part` to keep the continuity.
- This sentence "As the `::part()` pseudo-element is only visible to the parent DOM, when a shadow tree is nested, the parts are not visible to any ancestors other than the direct parent." is pretty long. I suggest we split it.
- Truncating the last sentence to avoid repetition with the next para
```suggestion
The global [`part`](/en-US/docs/Web/HTML/Global_attributes#part) attribute makes a shadow tree element visible to its parent DOM. Using the `part` name as the parameter, the {{CSSxRef("::part", "::part()")}} pseudo-element can then be used to apply CSS styles to elements in the shadow tree from outside of it. However, the `::part()` pseudo-element is only visible to the parent DOM. This means that when a shadow tree is nested, the parts are not visible to any ancestors other than the direct parent. The `exportparts` attribute solves this limitation.
```

> The `exportparts` attribute enables shadow tree parts to be visible outside the shadow DOM. This concept is referred to as "exporting". The `exportparts` attribute is placed on the element's _shadow host_, which is the element to which the _shadow tree_ is attached. The value of this attribute should be a comma-separated list of `part` names present in the shadow tree, which should be made available to the DOMs outside the current structure.

Reviewer: The last sentence can be simplified and made more clear
```suggestion
The `exportparts` attribute enables shadow tree parts to be visible outside the shadow DOM. This concept is referred to as "exporting". The `exportparts` attribute is placed on the element's _shadow host_, which is the element to which the _shadow tree_ is attached. The value of this attribute is a comma-separated list of `part` names present in the shadow tree. These names are made available to the DOMs outside the current structure.
```

The **`exportparts`** [global attribute](/en-US/docs/Web/HTML/Global_attributes) allows you to select and style elements existing in nested {{Glossary("shadow tree", "shadow trees")}}, by exporting their `part` names.

The shadow tree is an isolated structure where identifiers, classes, and styles cannot be reached by selectors or queries belonging to a regular DOM. There are two HTML attributes that can be applied to shadow tree elements that enable targeting CSS styles from outside to shadow tree: [`part`](/en-US/docs/Web/HTML/Global_attributes#part) and `exportparts`.

> The global attribute [`part`](/en-US/docs/Web/HTML/Global_attributes#part) makes a shadow tree element visible to its parent DOM. The `part` attribute is assigned to an element in the shadow tree with an identifier as its value. The {{CSSxRef("::part")}} pseudo-element containing that identifier can then be used to apply CSS styles to elements in the shadow tree from outside of the shadow tree.

Reviewer: This needs a bit more work
